### PR TITLE
Print an error if a collator connects to a node without real-overseer

### PR DESCRIPTION
**Pull request by tomaka**
It's not a lot of code, and it can prevent stupid accidental mistakes.

Note that I haven't tested if this works, but it seems pretty straight-forward.

### CODE REVIEW

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -649,6 +663,28 @@ pub fn new_full<RuntimeApi, Executor>(
...
+		let mut network_events = network.event_stream("dummy-collation-handler");
+		async move {
+			use futures::prelude::*;
+			while let Some(ev) = network_events.next().await {
+				if let sc_network::Event::NotificationStreamOpened { protocol, .. } = ev {
+					if protocol == "/polkadot/collation/1" {
+						tracing::warn!(
+							"Incoming collator on a node with parachains disabled. This warning \
+							is harmless and is here to warn developers that they might have
+							accidentally compiled their node without the parachains features
```
Comment (bkchr) [inline node/service/src/lib.rs:679]:
```suggestion
							accidentally compiled their node without the `real-overseer` feature
```

**Review by bkchr — APPROVED** [review]
TY!

**Review by ordian — APPROVED** [review]

**Comment by ordian** [comment]
I didn't observe a warning when running https://github.com/paritytech/polkadot/blob/master/parachain/test-parachains/adder/collator/README.md with validators w/o `real-overseer` feature. Is that expected?

**Comment by xlc** [comment]
Nice. This will partially address https://github.com/paritytech/polkadot/issues/2546

**Comment by tomaka** [comment]
> I didn't observe a warning when running https://github.com/paritytech/polkadot/blob/master/parachain/test-parachains/adder/collator/README.md with validators w/o real-overseer feature. Is that expected?

Good question.
Did you register the parachain as well?
Basically this warning would only be printed if we reach the state where collators actually try to collate.

**Comment by ordian** [comment]
Nice
```
2021-03-05 12:39:06  Incoming collator on a node with parachains disabled. This warning is harmless and is here to warn developers that they might have
							accidentally compiled their node without the parachains features
							enabled.
```

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -649,6 +663,28 @@ pub fn new_full<RuntimeApi, Executor>(
...
+	task_manager.spawn_handle().spawn("dummy-collation-handler", {
+		let mut network_events = network.event_stream("dummy-collation-handler");
+		async move {
+			use futures::prelude::*;
+			while let Some(ev) = network_events.next().await {
+				if let sc_network::Event::NotificationStreamOpened { protocol, .. } = ev {
+					if protocol == "/polkadot/collation/1" {
+						tracing::warn!(
+							"Incoming collator on a node with parachains disabled. This warning \
+							is harmless and is here to warn developers that they might have
```
Comment (ordian) [inline node/service/src/lib.rs:678]: you probably meant to use `\` after everyline? but that's fine as well

Comment (tomaka) [reply node/service/src/lib.rs:678]: Yeah

**Pull request merged**